### PR TITLE
Feature/sync flexbe

### PR DESCRIPTION
**Pull request by fmessmer**
the maintenance of `flexbe_behavior_engine` has switched from `team-vigir` to `FlexBe` organization :exclamation: 

there are a few changes on https://github.com/FlexBE/flexbe_behavior_engine/tree/noetic branch that seem useful for us, i.e. mainly:
- https://github.com/FlexBE/flexbe_behavior_engine/commit/4fd5bd4f2fa2a3a61c0e768496073ba2c248545e
- https://github.com/FlexBE/flexbe_behavior_engine/commit/bfefaa835eede9d77d81660e2eb841e9df5ab44f
- https://github.com/FlexBE/flexbe_behavior_engine/commit/ae7289e1092adb4e778714a978d42fb4caa9af35
- https://github.com/FlexBE/flexbe_behavior_engine/commit/5e52d0776368f1107c362945d015f88cbf7f1e34
- https://github.com/FlexBE/flexbe_behavior_engine/commit/2402618606002bcab9a4bbcf457a15b0a2cee12b

@benmaidel could you take a look?

### CODE REVIEW

File: flexbe_states/src/flexbe_states/subscriber_state.py [hunk 1]
```diff
@@ -53,8 +54,11 @@ def on_enter(self, userdata):
             self._sub.remove_last_msg(self._topic)
 
     def _connect(self):
-        msg_type, msg_topic, _ = rostopic.get_topic_class(self._topic)
-        if msg_topic == self._topic:
+        global_topic = self._topic
+        if global_topic[0] != '/':
+            global_topic = rospy.get_namespace() + global_topic
+        msg_type, msg_topic, _ = rostopic.get_topic_class(global_topic)
+        if msg_topic == global_topic:
```
Comment (LoyVanBeek) [inline flexbe_states/src/flexbe_states/subscriber_state.py:61]: This looks fine, but there could be some places where we already have something like this done ourselves. 
Cannot judge ATM. Maybe `grep` on `rospy.get_namespace` in our own code so see if there are any uses

File: flexbe_core/src/flexbe_core/proxy/proxy_subscriber_cached.py [hunk 2]
```diff
@@ -134,7 +134,9 @@ def has_msg(self, topic):
         @type topic: string
         @param topic: The topic of interest.
         """
-        return ProxySubscriberCached._topics[topic]['last_msg'] is not None
+        if self.is_available(topic):
+            return ProxySubscriberCached._topics[topic]['last_msg'] is not None
+        return False
```
Comment (LoyVanBeek) [inline flexbe_core/src/flexbe_core/proxy/proxy_subscriber_cached.py:139]: Also makes sense, I already expected things to work this way I think

**Review by LoyVanBeek — COMMENTED** [review]
Nothing springs out as something that might potentially break our stuff, just scrolling through the code. Some method & State added that we don't yet use, some additional logging, that can't hurt us. 

The other changes seem sensible and I don't think they'll bite us. We might however check if some of the fixes make some of our own obsolete.

**Comment by fmessmer** [comment]
this has been running on robots and emulation for a while now...guess it's safe to merge

**Pull request merged**